xdsl/interpreters/eqsat_pdl_interp.py

```diff
@@ -1,26 +1,42 @@
 from __future__ import annotations
 
 from dataclasses import dataclass, field
-from typing import Any
+from typing import Any, cast
 
 from xdsl.dialects import eqsat, pdl_interp
 from xdsl.dialects.builtin import ModuleOp
 from xdsl.dialects.pdl import ValueType
 from xdsl.interpreter import (
     Interpreter,
+    ReturnedValues,
+    Successor,
     impl,
+    impl_terminator,
     register_impls,
 )
 from xdsl.interpreters.pdl_interp import PDLInterpFunctions
-from xdsl.ir import Operation, OpResult
+from xdsl.ir import Block, Operation, OpResult, SSAValue
 from xdsl.transforms.common_subexpression_elimination import KnownOps
 from xdsl.utils.disjoint_set import DisjointSet
 from xdsl.utils.exceptions import InterpretationError
+from xdsl.utils.scoped_dict import ScopedDict
+
+
+@dataclass
+class BacktrackPoint:
+    block: Block
+    block_args: tuple[SSAValue, ...]
+    scope: ScopedDict[SSAValue, Any]
+    gdo_op: pdl_interp.GetDefiningOpOp
+    index: int
+    max_index: int
```

Comment:
Can you please add doc comments for the class, and each of the fields? I think more is better for this tricky part of the mechanism.